wire-compiler/src/test/java/com/squareup/wire/WireCompilerTest.java

```diff
@@ -138,18 +139,29 @@ private void testProtoWithRegistry(String[] sources, String registryClass, Strin
 
   private void testProtoWithRoots(String[] sources, String roots, String[] outputs)
       throws Exception {
+    String[] extraArgs = {};
+    this.testProtoWithRoots(sources, roots, outputs, extraArgs);
+  }
+
+  private void testProtoWithRoots(String[] sources, String roots, String[] outputs,
+      String[] extraArgs)
+      throws Exception {
     int numFlags = 4;
```

Comment:
I followed the pattern of other tests, but note that we don't need to invoke the compiler with an array of strings if we don't want to, we could construct a CommandLineOptions instance.  But it would be a bit awkward and require us to pass in null or empty list/set arguments.   I think a better programmatic interface would be to be able to initialize CommandLineOptions with builder pattern.